File: src/main/java/io/github/alexiscomete/lapinousecond/commands/CommandWithAccount.java
package io.github.alexiscomete.lapinousecond.commands;

import io.github.alexiscomete.lapinousecond.entity.Player;
import org.javacord.api.event.message.MessageCreateEvent;

public abstract class CommandWithAccount extends CommandBot {
    public CommandWithAccount(String description, String name, String totalDescription, String... perms) {
        super(description, name, totalDescription, perms);
    }

    @Override
    public void execute(MessageCreateEvent messageCreateEvent, String content, String[] args) {
        Player p = saveManager.players.get(messageCreateEvent.getMessageAuthor().getId());
        if (p == null) {
            messageCreateEvent.getMessage().reply("Vous devez d'abord vous créer un compte avec -start");
        } else {
            execute(messageCreateEvent, content, args, p);
        }
    }

    public abstract void execute(MessageCreateEvent messageCreateEvent, String content, String[] args, Player p);
}

File: src/main/java/io/github/alexiscomete/lapinousecond/commands/CommandWithAccount.kt
package io.github.alexiscomete.lapinousecond.commands

import io.github.alexiscomete.lapinousecond.entity.Player
import io.github.alexiscomete.lapinousecond.entity.players
import org.javacord.api.event.message.MessageCreateEvent

abstract class CommandWithAccount(
    description: String,
    name: String,
    totalDescription: String,
    vararg perms: String
) : CommandBot(
    description, name, totalDescription, *perms
) {
    override fun execute(messageCreateEvent: MessageCreateEvent, content: String, args: Array<String>) {
        val p = players[messageCreateEvent.messageAuthor.id]
        p?.let { execute(messageCreateEvent, content, args, it) }
            ?: messageCreateEvent.message.reply("Vous devez d'abord vous créer un compte avec -start")
    }

    abstract fun execute(messageCreateEvent: MessageCreateEvent, content: String, args: Array<String>, p: Player)
}